docs/how-to/hip_runtime_api/memory_management/virtual_memory.rst

```diff
@@ -0,0 +1,154 @@
+.. meta::
+  :description: This chapter describes introduces Virtual Memory (VM) and shows
+                how to use it in AMD HIP.
+  :keywords: AMD, ROCm, HIP, CUDA, virtual memory, virtual, memory, UM, APU
+
+.. _virtual_memory:
+
+********************************************************************************
+Virtual memory management
+********************************************************************************
+
+Memory management is important when creating high-performance applications in
+the HIP ecosystem. Both allocating and copying memory can result in bottlenecks,
+which can significantly impact performance.
+
+Global memory allocation in HIP uses the C language style allocation function.
+This works fine for simple cases but can cause problems if your memory needs
+change. If you need to increase the size of your memory, you must allocate a
+second larger buffer and copy the data to it before you can free the original
+buffer. This increases overall memory usage and causes unnecessary ``memcpy``
+calls. Another solution is to allocate a larger buffer than you initially need.
+However, this isn't an efficient way to handle resources and doesn't solve the
+issue of reallocation when the extra buffer runs out.
+
+Virtual memory management solves these memory management problems. It helps to
+reduce memory usage and unnecessary ``memcpy`` calls.
+
+.. _memory_allocation_virtual_memory:
+
+Memory allocation
+================================================================================
+
+Standard memory allocation uses the :cpp:func:`hipMalloc` function to allocate a
+block of memory on the device. However, when using virtual memory, this process
+is separated into multiple steps using the :cpp:func:`hipMemCreate`,
+:cpp:func:`hipMemAddressReserve`, :cpp:func:`hipMemMap`, and
+:cpp:func:`hipMemSetAccess` functions. This guide explains what these functions
+do and how you can use them for virtual memory management.
+
+Allocate physical memory
+--------------------------------------------------------------------------------
+
+The first step is to allocate the physical memory itself with the
+:cpp:func:`hipMemCreate` function. This function accepts the size of the buffer,
+an ``unsigned long long`` variable for the flags, and a
+:cpp:struct:`hipMemAllocationProp` variable. :cpp:struct:`hipMemAllocationProp`
+contains the properties of the memory to be allocated, such as where the memory
+is physically located and what kind of shareable handles are available. If the
+allocation is successful, the function returns a value of
+:cpp:enumerator:`hipSuccess`, with :cpp:type:`hipMemGenericAllocationHandle_t`
+representing a valid physical memory allocation. The allocated memory size must
```

Comment:
```suggestion
representing a valid physical memory allocation. 

The allocated memory size must
```

I would at least separate these last two sentences out of the preceding paragraph. I might also suggest making them a Note perhaps? Finally, I think it needs to be more definite about who needs to align the memory size with the granularity... _You_ must align the memory size...